server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -351,7 +397,18 @@ public SortFieldAndFormat build(QueryShardContext context) throws IOException {
                     && (sortMode == SortMode.SUM || sortMode == SortMode.AVG || sortMode == SortMode.MEDIAN)) {
                 throw new QueryShardException(context, "we only support AVG, MEDIAN and SUM on number based fields");
             }
-            SortField field = fieldData.sortField(missing, localSortMode, nested, reverse);
+            final SortField field;
+            if (numericType != null) {
+                if (fieldData instanceof SortedNumericDVIndexFieldData == false) {
```

Comment:
why do you check against this class rather than IndexNumericFieldData?

Reply:
This is a mistake, I pushed https://github.com/elastic/elasticsearch/pull/38095/commits/24c961afd1d7b6e0197d8d15e015761ffeae6d7d